### ISSUE TITLE
Consider adoption of xarray DataTree as intermediate container

### ISSUE THREAD

**Issue by djhoese** [NRLMMD-GEOIPS/geoips#318]
# Requested Update

### Description

Xarray DataTree is a new but actively developed Python library that is intended to be merged upstream into xarray in the future. In the most basic sense it acts as an xarray Dataset of xarray Datasets that can constructs a "tree" of Datasets/DataArrays and allows for multiple levels of nodes. I had mentioned this in the slack during the workshop, but wanted to bring it up here in case an asynchronous discussion could be started before the weekly meeting.

https://xarray-datatree.readthedocs.io/en/latest/

### Background and Motivation

From what I saw at the meeting/workshop last week, GeoIPS prefers readers to return a `dict` of `xarray.Dataset` objects (you call these "xarrays", but I assumed that term was being used for "DataArrays). Additionally you have time information and arbitrary/useful metadata. This structure and use case seems to fit in very well with the DataTree design. Satpy has some draft pull requests for converting to a DataTree structure but we've only just starting scratching the surface. I'm interested in how a separate library (geoips) with separate use cases sees this type of functionality, especially if you see any major downsides or missing functionality.

### Alternative Solutions

* Keep your existing structure/design
* Adopt something complex like Satpy's `DataID`s

**Comment by mindyls** [comment]
I looked into DataTrees a bit (@lambertaa suggested it as well), and it definitely looks relevant.  Worth looking into adopting, we can discuss more next Tuesday morning!

**Comment by evrose54** [comment]
If we end up implementing DataTrees, we need to make sure to update the sphinx documentation, specifically under the 'Developer Guide' --> 'Xarray and NetCDF Metadata Standards' Section. It looks good now, but this should be noted if we make these changes in the future.

**Comment by djhoese** [comment]
Yes. Thinking about that topic and even exporting to NetCDF/CF, we may end up with shared utilities between satpy and geoips for converting between a DataTree and a single Dataset object. I assume GeoIPS tries to use CF standards where it makes sense. Satpy has had conversions functions to easily create a CF compatible xarray.Dataset object from our collection of DataArrays. The more that DataTree pops up in conversations the more pull requests and CF-related things are adapting conversions from DataTree to Datasets to "collapse" the tree into a single NetCDF-able object.
